### PR TITLE
ci: enable use of local fuzz introspector webapp

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Adds logic for running experiments using a local deployment of the introspector webapp. This can (1) help with network failures and (2) make it possible to run experiments if for some reason introspector.oss-fuzz.com is down.

Also adds a small benchmark directory with a single project which is meant for PRs that don't need full experiments to run, but where it would still be nice to test a full run to make sure the infrastructure doesn't break.

### CODE REVIEW

**Comment by DavidKorczynski** [comment]
This supercedes https://github.com/google/oss-fuzz-gen/pull/305

**Comment by DavidKorczynski** [comment]
/gcbrun exp -n dk-test-infra140 -m vertex_ai_gemini-1-5 -b minor-for-ci -i

**Comment by DavidKorczynski** [comment]
/gcbrun exp -n dk-test-infra141 -m vertex_ai_gemini-1-5 -b minor-for-ci -i

**Comment by DavidKorczynski** [comment]
/gcbrun exp -n dk-test-infra142 -m vertex_ai_gemini-1-5 -b minor-for-ci -i

**Comment by DavidKorczynski** [comment]
Experiment is looking good

File: report/docker_run.sh [hunk 1]
```diff
@@ -85,6 +86,60 @@ then
   echo "DELAY was not specified as the sixth argument. Defaulting to ${DELAY:?}."
 fi
 
+launch_local_introspector_deployment() {
```
Comment (DonggeLiu) [inline report/docker_run.sh:89]: This function is quite long and independent, would it be better to do this in a separate file like `upload_report.sh`?

File: report/docker_run.sh [hunk 2]
```diff
@@ -112,12 +167,19 @@ $PYTHON run_all_experiments.py \
...
 export ret_val=$?
 
 touch /experiment_ended
 
+if [[ "$USE_LOCAL_INTROSPECTOR" = "true" ]]
+then
+  echo "Shutting down introspector"
+  curl --silent http://localhost:8080/api/shutdown || true
+fi
+
```
Comment (DonggeLiu) [inline report/docker_run.sh:182]: Maybe move to the same new file as `launch_local_introspector_deployment`?

File: ci/request_pr_exp.py [hunk 3]
```diff
@@ -250,6 +255,8 @@ def _fill_template(args: argparse.Namespace) -> str:
   exp_env_vars['GKE_EXP_NAME'] = args.experiment_name
   exp_env_vars['GKE_EXP_REQ_CPU'] = args.request_cpus
   exp_env_vars['GKE_EXP_REQ_MEM'] = f'{args.request_memory}Gi'
+  exp_env_vars[
+      'GKE_EXP_LOCAL_INTROSPECTOR'] = 'true' if args.local_introspector else ''
```
Comment (DonggeLiu) [inline ci/request_pr_exp.py:259]: nit: This takes two lines anyway.
```python
if args.local_introspector:
  exp_env_vars['GKE_EXP_LOCAL_INTROSPECTOR'] = 'true'
```

**Review by DonggeLiu — APPROVED** [review]

File: ci/request_pr_exp.py (hunk 3 above)
Comment (DavidKorczynski) [reply ci/request_pr_exp.py]: done

File: report/docker_run.sh (hunk 2 above)
Comment (DavidKorczynski) [reply report/docker_run.sh:137]: We can't -- this has to be called after the experiment so we should call it from the one that controls the experiment running

**Comment by DavidKorczynski** [comment]
/gcbrun exp -n dk-test-infra142 -m vertex_ai_gemini-1-5 -b minor-for-ci

**Comment by DavidKorczynski** [comment]
/gcbrun exp -n dk-test-infra5000 -m vertex_ai_gemini-1-5 -b minor-for-ci

**Comment by DavidKorczynski** [comment]
/gcbrun exp -n dk-test-infra5001 -m vertex_ai_gemini-1-5 -b minor-for-ci -i

File: report/docker_run.sh (hunk 1 above)
Comment (DavidKorczynski) [reply report/docker_run.sh]: Done, thanks!

File: report/docker_run.sh (hunk 2 above)
Comment (DonggeLiu) [reply report/docker_run.sh:137]: Would this trick work?
https://github.com/google/oss-fuzz-gen/blob/main/report/upload_report.sh#L102-L106

E.g.,
```bash
while [[ ! -f /experiment_ended ]]; do
  sleep 600
done

# Stop server now.
```

**Pull request merged**